src/schema/browser-apis.js

```diff
@@ -59,18 +160,47 @@ export function isDeprecatedApi(namespace, property) {
   );
 }
 
-export function hasBrowserApi(namespace, property) {
-  const schema = schemas[namespace];
+export function hasBrowserApi(
+  namespace,
+  property,
+  addonMetadata,
+  apiSchemas = schemas
+) {
+  const schema = apiSchemas[namespace];
+
   // We "have" the API if it's deprecated or temporary so we don't double warn.
   if (
     isTemporaryApi(namespace, property) ||
-    isDeprecatedApi(namespace, property)
+    isDeprecatedApi(namespace, property, addonMetadata, apiSchemas)
   ) {
     return true;
   }
 
   const schemaItem =
     getObjectProperty(schema, property) || getArrayProperty(schema, property);
 
+  const manifestVersion = getManifestVersion(addonMetadata);
+  const min_manifest_version = getMinManifestVersion(
+    namespace,
+    property,
+    addonMetadata,
+    apiSchemas
+  );
+  const max_manifest_version = getMinManifestVersion(
```

Comment:
Shouldn't that be `getMaxManifestVersion()`?

Reply:
I confirm that this was the reason behind the unexpected UNSUPPORTED_API linter warnings that you did mention in https://github.com/mozilla/addons-linter/pull/3633#issuecomment-831129468 

As I was also guessing in https://github.com/mozilla/addons-linter/pull/3633#issuecomment-831140404 , there was also one missing test case or assertion to cover that scenario explicitly (I have applied both locally in a fixup commit that I'm going to push as part of the next update on this PR).